File: Backend/Environment/EnvironmentAssembly/assemblyEnvironment/assemblyEnvironmentVars.Class.vb
Imports System.Drawing
Imports System.Windows.Forms

Public Class assemblyEnvironmentVarsClass

    Public Property MenuPanelBackColor As Color
    Public Property menu As assemblyMenuClass




    '' ???????
    Structure subMenuDesignPropertiesClass
        Public Property height As Integer
        Public Property width As Integer
        Public Property widthClosed As Integer
        Public Property bkColor As Color
        Public Property border As Boolean
    End Structure






    Public Class assemblyMenuClass
        Public Property menuItem As New List(Of assemblymenuItemClass)
        Public Property menuSort As New List(Of Integer)
    End Class

    Public Class assemblymenuItemClass
        'settings for loading contents
        Public Property nameSpaceString As String         ' Note that I´m in namespace  "ConsoleApplication1.MyClassA"
        Public Property assemblyFilename As String
        Public Property contentToLoadForm As System.Windows.Forms.Form
        Public Property showAsDialog As Boolean

        'settings for executing internal code tasks
        Public Property tasks As New List(Of String)

        'settings for design 
        Public Property menuUID As String
        Public Property menuTitle As String               ' menu title string for translations
        Public Property menuIndex As Integer              '1 is the first on TOP; 0 is disabled
        Public Property subMenuIndex As Integer = -1      '1 is the first on TOP; false means is a menu 
        Public Property menuDesignProperties As subMenuDesignPropertiesClass

        Public Property notifications As Integer          ' number of notification pending on menu item
        Public Property icon As String

        Public Property subMenuPanel As Panel
        Public Property iconPicHolder As List(Of PictureBox)

        'menu wrapper
        Public Property menuWrapperPanel As Panel
        Public Property state As Boolean
        Public Property menuWrapperOpenHeight As Integer
    End Class

End Class

File: Backend/Environment/EnvironmentAssembly/EnvironmentAssembly/assemblyEnvironmentVars.Class.cs
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

namespace AeonLabs.Environment.Assembly
{
    public class assemblyEnvironmentVarsClass
    {
        public Color MenuPanelBackColor { get; set; }
        // duplicate on layout env
        public menuClass menu { get; set; } = new menuClass();

        public class menuClass
        {
            public List<menuItemClass> items { get; set; } = new List<menuItemClass>(); // menu items
            public List<int> menuSort { get; set; } = new List<int>();
            public menuDesignPropertiesClass properties { get; set; } = new menuDesignPropertiesClass();

            public class menuDesignPropertiesClass
            {
                public int height { get; set; }
                public int width { get; set; }
                public int ClosedStateSize { get; set; }
                public Color backColor { get; set; }
                public bool border { get; set; }
            }
        }

        public class menuItemClass
        {
            // settings for loading contents
            public string nameSpaceString { get; set; }         // Note that I´m in namespace  "ConsoleApplication1.MyClassA"
            public string assemblyFilename { get; set; }
            public Form formWithContentsToLoad { get; set; }
            public bool showAsDialog { get; set; }

            // settings for executing internal code tasks
            public List<string> tasks { get; set; } = new List<string>();

            // settings for design 
            public string menuUID { get; set; }
            public string menuTitle { get; set; }               // menu title string for translations
            public int menuIndex { get; set; }              // 1 is the first on TOP; 0 is disabled
            public int subMenuIndex { get; set; } = -1;      // 1 is the first on TOP; false means is a menu 
            public int menuListIndex { get; set; }            // index of the TOP menu item within the menuitems list
            public menuClass.menuDesignPropertiesClass menuDesignProperties { get; set; }
            public int notifications { get; set; }          // number of notification pending on menu item
            public string icon { get; set; }
            public Panel menuItemPanel { get; set; }
            public List<PictureBox> iconPicHolder { get; set; }

            // menu wrapper
            public Panel menuWrapperPanel { get; set; }
            public bool isOpen { get; set; }
            public int menuWrapperOpenHeight { get; set; }
        }
    }
}